app/src/main/java/org/wordpress/pioupiou/postlist/PostListActivity.java

```diff
@@ -124,17 +200,100 @@ public void onDismissed() {
         sendButton.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View v) {
+                mNewPostContent = editText.getText().toString();
                 // Clear the saved text
                 editText.setText("");
                 tooltip.dismiss(true);
-                publishPost(editText.getText().toString());
+                publishPost();
             }
         });
     }
 
     // Publish post
-    private void publishPost(String text) {
-        Timber.i("Publishing post: " + text);
-        // TODO: Prepare the post and dispatch the action
+    private void publishPost() {
+        if (TextUtils.isEmpty(mNewPostContent)) {
+            ToastUtils.showToast(this, "Can't publish an empty post");
+            return;
+        }
+
+        // instantiate a new post, event handler will take care of setting the content and uploading it
+        PostStore.InstantiatePostPayload payload = new PostStore.InstantiatePostPayload(getSite(), false);
+        mDispatcher.dispatch(PostActionBuilder.newInstantiatePostAction(payload));
+    }
+
+    private boolean hasPostFragment() {
+        return getPostFragment() != null;
+    }
+
+    private PostListFragment getPostFragment() {
```

Comment:
Nitpacking, but this and `hasPostFragment()` may as well be called `get/hasPostListFragment()` for consistency.